shards/shards.go

```diff
@@ -150,17 +153,97 @@ type shardedSearcher struct {
 	ranked        []rankedShard
 
 	priority map[string]float64
+
+	// public is a set of public repository names.
+	public map[string]struct{}
```

Comment:
Similarly to `indexData` we could use a `roaring.Bitmap` of Sourcegraph IDs here to save space.

Reply:
I agree that it would be better to store IDs here. However, to store IDs, we have to update public.txt to store IDs as well. I like that public.txt contains clear repo names. We can easily see which repos cause trouble by intersecting various lists (should be indexed, is indexed, public.txt). To not make things harder we should migrate all of Zoekt from names to ids, which probably should be a project by itself. 

With regards to the size of this map: Currently, public.txt is 1.2mb on disk. We store this map only once per instance, so the impact is minimal.

@keegancsmith WDYT?

Reply:
SGTM